### PR TITLE
[MBL-17881][All] Remove Class transformers

### PR DESCRIPTION
**Pull request by tamaskozmer**
The ticket was about handling locales in the Parent app, but that was implemented by a class transformer that we didn't want to reuse in the Parent app due to deprecation in the AGP so I implemented a different solution. Since I was already doing this I replaced all the class transformers with the same approach. So we should extensively test all the features that relied on class transformers.

Test plan: 
- Verify that locale changes are reflected in the Parent app (locale needs to be changed in Canvas user/admin settings).
- Verify that Masquerading works correctly in the Student/Teacher apps.
- Verify that locales can be selected in the settings of the Student app when in debug mode.
- Verify that Heap screenviews are sent correctly as before.
- Verify that Pandata events are sent correctly as before.

refs: MBL-17881
affects: All
release note: none

## Checklist

- [x] Run E2E test suite
- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by domonkosadam — CHANGES_REQUESTED** [review]
QA + 1

File: apps/teacher/src/main/java/com/instructure/teacher/adapters/AttendanceListRecyclerAdapter.kt [hunk 1]
```diff
@@ -25,12 +25,12 @@ import com.instructure.teacher.databinding.AdapterAttendanceBinding
...
 import com.instructure.teacher.viewinterface.AttendanceListView
-import instructure.androidblueprint.SyncPresenter
-import instructure.androidblueprint.SyncRecyclerAdapter
+import com.instructure.pandautils.blueprint.SyncPresenter
+import com.instructure.pandautils.blueprint.SyncRecyclerAdapter
 
 class AttendanceListRecyclerAdapter(
     mContext: Context,
-    presenter: SyncPresenter<Attendance, AttendanceListView>,
+    presenter: com.instructure.pandautils.blueprint.SyncPresenter<Attendance, AttendanceListView>,
```
[the remark this answers is not in the document]
Comment (tamaskozmer) [reply apps/teacher/src/main/java/com/instructure/teacher/adapters/AttendanceListRecyclerAdapter.kt:33]: I moved around some classes and probably that's causing these. I removed a couple of these, but seems like I missed some so I will fix these.

File: libs/pandautils/src/main/java/com/instructure/pandautils/base/AppConfigProvider.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+package com.instructure.pandautils.base
+
+import android.app.Activity
+
+data class AppConfig(
+    val appName: String,
```
[the remark this answers is not in the document]
Comment (tamaskozmer) [reply libs/pandautils/src/main/java/com/instructure/pandautils/base/AppConfigProvider.kt:21]: We still need the appName as a string for the ScreenViews, but I can convert this to an enum.

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/pageview/PageViewAnnotations.kt [hunk 3]
```diff
@@ -17,21 +17,17 @@
 package com.instructure.canvasapi2.utils.pageview
 
 @Target(AnnotationTarget.CLASS)
-@Retention(AnnotationRetention.BINARY)
+@Retention(AnnotationRetention.RUNTIME)
```
[the remark this answers is not in the document]
Comment (tamaskozmer) [reply libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/pageview/PageViewAnnotations.kt:20]: We tried different solutions in the past that manipulates the code in compile time, but couldn't make it work

**Review by domonkosadam — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by hermannakos — APPROVED** [review]

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**